src/Symfony/Component/Routing/Route.php

```diff
@@ -563,7 +563,7 @@ private function sanitizeRequirement(string $key, string $regex)
 
         if ('$' === substr($regex, -1)) {
             $regex = substr($regex, 0, -1);
-        } elseif (\strlen($regex) - 2 === strpos($regex, '\\z')) {
```

Comment:
Unrelated to the PR: couldn't this be written as this?
`} elseif (false !== strpos($regex, '\\z', -2)) {`

Reply:
@nicolas-grekas tried `strpos` and `strrpos`:

`strpos(): Offset not contained in string in `
`strrpos(): Offset is greater than the length of haystack string`